modulemd/v2/tests/ModulemdTests/merger.py

```diff
@@ -0,0 +1,177 @@
+
+        self.assertEqual(baseline, deduplicated_idx.dump_to_string())
+
+    def test_merger(self):
+        # Get a set of objects in a ModuleIndex
+        base_index = Modulemd.ModuleIndex()
+        base_index.update_from_file(
+            path.join(
+                self.source_root,
+                "modulemd/v2/tests/test_data/merging-base.yaml"))
+
+        # Baseline
+        httpd_defaults = base_index.get_module('httpd').get_defaults()
+        self.assertIsNotNone(httpd_defaults)
+        self.assertEqual(httpd_defaults.get_default_stream(), '2.2')
+        httpd_profile_streams = httpd_defaults.get_streams_with_default_profiles()
+        self.assertEqual(len(httpd_profile_streams), 2)
+        self.assertTrue('2.2' in httpd_profile_streams)
+        self.assertTrue('2.8' in httpd_profile_streams)
+        self.assertEqual(
+            len(httpd_defaults.get_default_profiles_for_stream('2.2')), 2)
+        self.assertTrue(
+            'client' in httpd_defaults.get_default_profiles_for_stream('2.2'))
+        self.assertTrue(
+            'server' in httpd_defaults.get_default_profiles_for_stream('2.2'))
+        self.assertTrue(
+            'notreal' in httpd_defaults.get_default_profiles_for_stream('2.8'))
+
+        self.assertEqual(
+            httpd_defaults.get_default_stream('workstation'), '2.4')
+        httpd_profile_streams = httpd_defaults.get_streams_with_default_profiles(
+            'workstation')
+        self.assertEqual(len(httpd_profile_streams), 2)
+        self.assertTrue('2.4' in httpd_profile_streams)
+        self.assertTrue('2.6' in httpd_profile_streams)
+
+        self.assertEqual(
+            len(httpd_defaults.get_default_profiles_for_stream('2.4', 'workstation')), 1)
+        self.assertEqual(
+            len(httpd_defaults.get_default_profiles_for_stream('2.6', 'workstation')), 3)
+
+        # Get another set of objects that will override the above
+        override_index = Modulemd.ModuleIndex()
+        override_index.update_from_file(
+            path.join(
+                self.source_root,
+                "modulemd/v2/tests/test_data/overriding.yaml"))
+
+        # Test that adding both of these at the same priority level fails
+        # with a merge conflict.
+        merger = Modulemd.ModuleIndexMerger()
+        merger.associate_index(base_index, 0)
+        merger.associate_index(override_index, 0)
+
+        with self.assertRaisesRegex(GLib.GError, 'Module stream mismatch in merge'):
+            merged_index = merger.resolve()
+
+        # Test that override_index at a higher priority level succeeds
+        # Test that adding both of these at the same priority level fails
+        # with a merge conflict.
+        # Use randomly-selected high and low values to make sure we don't have
+        # sorting issues.
+        merger = Modulemd.ModuleIndexMerger()
+        merger.associate_index(base_index, random.randint(1, 100))
```

Comment:
Perhaps it's an idea to print these indexes, so that reproducing the test is simple?

Reply:
I can do that, sure. When I get around to writing the matching test in C, the GLib test library provides a mechanism for doing this as well (see the example in the 1.x code).

Reply:
In Python unittests, you can just `print()` them, and they'll only show if the test failed.